pyproject.toml

```diff
@@ -42,7 +42,8 @@ dependencies = [
     "anytree>=2.8.0",
     "sympy>=1.12",
     "typing-extensions>=4.10.0",
-    "pandas>=1.5.0"
+    "pandas>=1.5.0",
+    "pooch>=1.8.1",
```

Comment:
No worries on keeping it as an optional dependency too, but can you specify which previously existent file? Feel free to resolve this conversation further.

Reply:
Ah, for example, earlier to this PR one could do

```
drive_cycle = pd.read_csv(
    "pybamm/input/drive_cycles/US06.csv", comment="#", header=None
).to_numpy()
```

in a drive cycle simulation to retrieve a NumPy array from the CSV-converted `DataFrame` object – but, after this PR, one needs to use `data_loader.get_data("US06.csv")` to access it. We have added an entry about this in the CHANGELOG, which is fine, but we would also like people to receive functionalities to work with these files without having to install any additional optional dependencies such as `pybamm[data]` after they try to upgrade their workflows to v24.5. 

(keeping it optional would lead to another error about `pooch` not being installed, which could be disorienting to unsuspecting users who are not reading the CHANGELOG as often)